packages/typescript-estree/jest.config.js

```diff
@@ -1,9 +1,19 @@
 'use strict';
-
 // @ts-check
+
+console.log(
+  'Running with',
+  process.version,
+  'which startsWith:',
+  process.version.startsWith('14'),
+);
+
 /** @type {import('@jest/types').Config.InitialOptions} */
 module.exports = {
   ...require('../../jest.config.base.js'),
+  testPathIgnorePatterns: process.version.startsWith('14')
+    ? ['/node_modules/', '/parse\\.moduleResolver/']
+    : ['/node_modules/'],
```

Comment:
😬 I had hoped that it was just Node 14 running out of memory, but no - 18 runs out too. https://github.com/typescript-eslint/typescript-eslint/actions/runs/5206328809/jobs/9392825081

```plaintext
Running with v18.16.0 which startsWith: false
```

```
Start of describe: parse.moduleResolver.placeholder-error.test.ts

      at log (tests/lib/parse.moduleResolver.placeholder-error.test.ts:13:11)
<--- Last few GCs --->
[2965:0x6b05f90]   159828 ms: Mark-sweep (reduce) 2040.3 (2082.4) -> 2039.4 (2082.9) MB, 1581.5 / 0.0 ms  (average mu = 0.[261](https://github.com/typescript-eslint/typescript-eslint/actions/runs/5206328809/jobs/9392825081#step:5:262), current mu = 0.180) allocation failure; scavenge might not succeed
```

Reply:
I think we're being hit by https://github.com/microsoft/TypeScript/issues/54491.